buildcheck/components/progress_tracker.py

```diff
@@ -0,0 +1,144 @@
+import reflex as rx
+
+class ProgressState(rx.State):
+    current_step: int = 2  # 1: Received, 2: Under Review, 3: Review Completed
+    
+    def set_step(self, step: int):
+        self.current_step = step
+
+def get_status(step_number: int, current_step: int) -> str:
```

Comment:
It seems like based on how reflex works, this function should maybe be a part of progress state, as a computed var. So that it can actually recompute when the state changes.

From my understanding anything not in state, and anything that doesn't use vars will not update dynamically after the code is compiled.

So it's possible right now that the component won't update properly once current_step changes.


We can honestly leave that as a bug to fix in a future PR.